common/tools/dev-tool/src/commands/admin/migrate-tests.ts

```diff
@@ -0,0 +1,201 @@
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function getRushJson(): Promise<any> {
+  if (_rushJson) return _rushJson;
+
+  const rushJsonText = await readFile(resolve(__dirname, "../../../../../../rush.json"), "utf-8");
+
+  return (_rushJson = JSON.parse(stripJsonComments(rushJsonText)));
+}
+
+/**
+ * The shape of a rush.json `projects` entry.
+ */
+export interface RushJsonProject {
+  /**
+   * The name of the package.
+   */
+  packageName: string;
+  /**
+   * The path to the project, relative to the monorepo root.
+   */
+  projectFolder: string;
+
+  /**
+   * The version policy name.
+   */
+  versionPolicyName: string;
+}
+
+export const commandInfo = makeCommandInfo(
+  "migrate-tests",
+  "migrates a package to the latest tests standards",
+  {
+    "package-name": {
+      description: "The name of the package to migrate",
+      kind: "string",
+      required: false,
+    },
+  },
+);
+
+const TEST_ESM_NODE = "dev-tool run test:vitest --esm";
+
+export default leafCommand(commandInfo, async ({ "package-name": packageName }) => {
+  let projectFolder: string;
+
+  if (!packageName) {
+    projectFolder = process.cwd();
+    const info = await resolveProject(process.cwd());
+    packageName = info.packageJson.name;
+  } else {
+    const root = await resolveRoot();
+
+    const rushJson = await getRushJson();
+    const projects = rushJson.projects;
+
+    const project = projects.find((p: RushJsonProject) => p.packageName === packageName);
+    if (!project) {
+      log.error(`Package ${packageName} not found in rush.json`);
+      return false;
+    }
+
+    projectFolder = resolve(root, project.projectFolder);
+  }
+
+  log.info(`Migrating package ${packageName}`);
+
+  const packageJsonPath = resolve(projectFolder, "package.json");
+  const packageJsonContent = await readFile(packageJsonPath, "utf-8");
+  const packageJson = JSON.parse(packageJsonContent);
+
+  if (packageJson.scripts) {
+    const unitTestNode = packageJson.scripts["unit-test:node"];
+    let unitTestBrowser = packageJson.scripts["unit-test:browser"];
+
+    // If it contains build:test, replace it
+    if (unitTestBrowser?.includes("npm run build:test")) {
+      log.info("Replacing build:test script");
+      unitTestBrowser = unitTestBrowser.replace("npm run build:test", "dev-tool run build-test");
+    }
+
+    if (unitTestNode) {
+      packageJson.scripts["test:node"] = unitTestNode;
+    }
+    if (unitTestBrowser) {
+      packageJson.scripts["test:browser"] = unitTestBrowser;
+      delete packageJson.scripts["unit-test:browser"];
+    }
+
+    // Save the integration test if it's not the same as the ESM node test
+    if (packageJson.scripts["integration-test:node"] !== TEST_ESM_NODE) {
+      log.info("Saving integration-test:node");
+      packageJson.scripts["test:node:live"] = packageJson.scripts["integration-test:node"];
+    }
+
+    // Clean up integration test scripts
+    delete packageJson.scripts["unit-test"];
```

Comment:
while we're doing this, we want to remove any references to `unit-test` and `integration-test` in the `eng` and `common` folders.

Example: https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-js+path%3A%2F%5Eeng%5C%2F%2F+integration-test&type=code

Reply:
Yes, that is the plan, let me make those changes now

Reply:
Fixed